Topic1/src/com/topic1/Counter.java

```diff
@@ -0,0 +1,26 @@
+package com.topic1;
+
+public class Counter {
+
+	private static Counter counter = null;
+	private int tNumber = 0;
+
+	private Counter() {
+
+	}
+
+	public static Counter getInstance() {
+
+		if (counter == null) {
+			return new Counter();
```

Comment:
[High] you must store this instance in your private *Counter* attribute.
Right now, you are creating a new instance each time that you call to *getInstance* method, because counter is *always* <code>null</code>